### ISSUE TITLE
Can't run on PS3

### ISSUE THREAD

**Issue by Bandud2** [Berny23/LD-ToyPad-Emulator#201]
For some reason I can't seem to get the emulator running when connected to my PS3. I ran the command to enable the automatic start up, and it works when connected to my computer, as I was able to connect it to my laptop, and was able to access the emulator from the browser without doing anything. When I connect it to my PS3 however, my raspberry pi does work ( It emits the green led) but I can't access the emulator from my browser. And yes, I am using a micro usb cable.

EDIT: I managed to get the emulator running when connected to the PS3 by simply connecting the usb to the right port of the raspberry pi ( The port closer to the edge). But the game still doesn't detect the emulator, and vice verca.

**Comment by Bandud2** [comment]
> What was solved: running pi or emulator (when connected to USB)?
> 
> Please give this: #188 a try. Should be easier to troubleshoot.

Both, the pi is running and the emulator works (I can connect to it with the browser). But it's not working ingame and I am getting the message to connect the toy pad to the console

**Comment by VladimirKuletski** [comment]
> > What was solved: running pi or emulator (when connected to USB)?
> > 
> > Please give this: #188 a try. Should be easier to troubleshoot.
> 
> Both, the pi is running and the emulator works (I can connect to it with the browser). But it's not working ingame and I am getting the message to connect the toy pad to the console

Please doublecheck the 3rd item from the guide - this is the most common issue. Either skip it or take from VM chapter

**Comment by Bandud2** [comment]
> > > What was solved: running pi or emulator (when connected to USB)?
> > > Please give this: #188 a try. Should be easier to troubleshoot.
> > 
> > 
> > Both, the pi is running and the emulator works (I can connect to it with the browser). But it's not working ingame and I am getting the message to connect the toy pad to the console
> 
> Please doublecheck the 3rd item from the guide - this is the most common issue. Either skip it or take from VM chapter

Do you mean the configuration step?

**Comment by Bandud2** [comment]
> What was solved: running pi or emulator (when connected to USB)?
> 
> Please give this: #188 a try. Should be easier to troubleshoot.

Ok so not only did this method not work, but now after I tried to install the thing like I did before, the node index.js command gives an error
![image_2024-07-31_164135832](https://github.com/user-attachments/assets/dfd6c9eb-0972-4595-8828-fee229942d53)

**Comment by Bandud2** [comment]
So after trying different fixes, I'm pretty much back at square 1. The emulator does start automatically when connected to my PS3, but the actual game doesn't detect it.

**Comment by Bandud2** [comment]
@Berny23 Do you think you can help me out with this? I'm honestly clueless as to why it isn't working

**Comment by Berny23** [comment]
usb_setup_script.sh creates the virtual device for USB communication. Your error code shows that the script didn't run properly.

**Comment by Bandud2** [comment]
> usb_setup_script.sh creates the virtual device for USB communication. Your error code shows that the script didn't run properly.

If you're talking about the ENOENT error then I already resolved it. I managed to get the emulator running on the pi, it just doesn't work in the game.

**Comment by Bandud2** [comment]
Ok nvm, I managed to fix my issue and it was honestly such a stupid one... Basically I was using  cable that, was capable of connecting the pi to the ps3 for the pi to be detected as a USB device, it didn't have enough power to do so.. Then I had another cable which was, funnily enough the exact opposite. It could supply the Pi with the power to actually run the emulator, but for some reason when I use it, my Pi is never detected. What I did was to connect the latter cable to the edge port of the pi, just for it to boot up and run the emulator, then connect the first cable to the "main" port so the game would actually recognize it. It's funny that if only I had a normal cable that had both capabilities, I wouldn't even have this issue...
